lib/proxy/archive_extract.rb

```diff
@@ -0,0 +1,21 @@
+module Proxy
+  class ArchiveExtract < Proxy::Util::CommandTask
+    include Util
+
+    def initialize(src, dst, skip_existing = true)
+
+      args = [which('7z')]
```

Comment:
7z is not commonly available. Is this really the only tool that can extract it? For example, it's not in CentOS 8 (which is the only EL platform now that we've dropped EL7). It is in EPEL, but at least right now Foreman is incompatible with EPEL8 (https://github.com/theforeman/foreman/pull/9317 may change that).

Looking at https://www.tecmint.com/extract-files-from-iso-files-linux/ there's also isoinfo (in the genisoimage package in EL8).

Also note that this should be reflected in packaging so that the command is actually available.

Reply:
Thanks for your idea with `isoinfo`. I tried to setup an alternative which makes use of it instead of `7z`. Unfortunately, the fact that `isoinfo` does not have an option to directly export the file makes some trouble:
It seems to be best practice with `isoinfo` to forward stdout to a file. AFAIK, this can't simply be adapted to the `CommandTask` class. It logs the stdout but I didn't find an easy way to store the stdout from [here](https://github.com/theforeman/smart-proxy/pull/838/files#diff-c2b0ceb586ecbff7db402cf8b53d18cec5ab6a5dac16e48ded1b194f34ac342cR32) to a file. 

I've created a "prototype" PR which summarizes my approach with `isoinfo` here: #838

Reply:
Changed to `isoinfo` implementation.